controlplane/pkg/prefix_pool/prefixpool.go

```diff
@@ -201,13 +205,38 @@ func ExtractPrefixes(prefixes []string, requests ...*connectioncontext.ExtraPref
 	if len(result) == 0 {
 		return nil, prefixes, fmt.Errorf("Failed to extract prefixes, there is no available %v", prefixes)
 	}
+
 	return result, newPrefixes, nil
 }
 
-func ExtractPrefix(prefixes []string, prefixLen uint32) (string, []string, error) {
+/* Check for excludedPrefixes intersection */
+func verifyExcluding(prefixes string, excludedPrefixes []string) error {
+	_, ipNet, err := net.ParseCIDR(prefixes)
+	if err != nil {
+		return err
+	}
+
+	excludedPrefixesPool, err := NewPrefixPool(excludedPrefixes...)
+	if err != nil {
+		return err
+	}
+
+	intersect, err := excludedPrefixesPool.Intersect(ipNet.String())
+	if err != nil {
+		return err
+	}
+
+	if intersect {
+		return fmt.Errorf("IPAM: Allocated IP intersects with excludedPrefix")
+	}
+
+	return nil
+}
+
+func ExtractPrefix(prefixes []string, prefixLen uint32, excludedPrefixes []string) (string, []string, error) {
```

Comment:
@lobkovilya Sorry for taking a bit longer to reply. I did try the test you suggested and although currently it is handled as a false negative (stating that it cannot allocate the desired network when in reality it is possible), digging deeper seems that the address allocation algorithm (the one with the subnet splitting) doesn't proceed to check the range from 10.20.1.0 to 10.20.255.255 as it should, but it ends at the 10.20.0.0 one.